### PR TITLE
Bug fix: Add remappings to solc options

### PR DESCRIPTION
**Pull request by eggplantzzz**
Previously, Truffle was not passing through remapping data to the Solidity compiler. This PR creates a default value for `compilers.solc.settings.remappings` in @truffle/config. It also passes the value on to the Solidity compiler at compile time in @truffle/compile-solidity so users can provide remappings in `truffle-config.js`.

Inspired by #2355

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -14271,6 +14271,53 @@ trim-right@^1.0.1:
   resolved "https://registry.yarnpkg.com/trim-right/-/trim-right-1.0.1.tgz#cb2e1203067e0c8de1f614094b9fe45704ea6003"
   integrity sha1-yy4SAwZ+DI3h9hQJS5/kVwTqYAM=
 
+truffle-blockchain-utils@^0.0.10:
```
Comment (gnidan) [inline yarn.lock]: What's up with these? Shouldn't they be the new names?

Comment (eggplantzzz) [reply yarn.lock]: Hehe, is it a dependency of a dependency?

Comment (eggplantzzz) [reply yarn.lock]: Yeah, I don't get it

Comment (eggplantzzz) [reply yarn.lock]: Well ethpm-registry has a dependency of truffle-contract.

Comment (gnidan) [reply yarn.lock]: ah that would do it!

**Pull request merged**